pom.xml

```diff
@@ -336,9 +336,31 @@
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
           <configuration>
+            <compilerId>javac-with-errorprone</compilerId>
+            <forceJavacCompilerUse>true</forceJavacCompilerUse>
             <source>1.8</source>
             <target>1.8</target>
+            <compilerArgs combine.children="override">
+              <compilerArg>-XepAllErrorsAsWarnings</compilerArg>
+              <!-- This would require adding ErrorProne annotations in our code: -->
```

Comment:
Fwiw, I'd be ok with that. Getting format validation is nice-to-have, and I don't see too many downsides to using the annotations. Totally fine with leaving that to later though.